sys/include/shell.h

```diff
@@ -48,6 +48,34 @@ extern "C" {
  */
 #define SHELL_DEFAULT_BUFSIZE   (128)
 
+/**
+ * @brief           Optional hook after readline has triggered.
+ * @details         User implemented function gets called after the shell
+ *                  readline is complete.
+ */
+void post_readline_hook(void);
```

Comment:
Shouldn't these functions be prefixed by the module name, e.g. `shell_post_readline_hook` or (what I would prefer) `shell_hook_post_readline`?

Reply:
Yup, namespacing is important, I will correct and directly squash. I currently want to use it for some paper evaluation stuff and am working on a module that can use this with other triggers to log the timings with instrumentation compensation.

Reply:
Just learned about the `trace` module. That is probably better to use.

Reply:
done